### PR TITLE
Alex/57 connect donation to firebase

### PR DESCRIPTION
**Pull request by alexandrashaw10**
#57

### CODE REVIEW

File: gradlew [hunk 1]
```diff
@@ -1,172 +0,0 @@
-#!/usr/bin/env sh
```
Comment (chiped) [inline gradlew]: we might need gradle wrapper for travis. let's keep it for now

File: app/src/main/java/com/example/una/DonationActivity.java [hunk 2]
```diff
@@ -26,48 +26,30 @@
...
-                    amountInput.addTextChangedListener(this);
-                }
-
-            }
-
-            @Override
-            public void afterTextChanged(Editable s) { }
-        });
+        CurrencyTextWatcher watcher = new CurrencyTextWatcher(amountInput, currentAmount);
+        amountInput.addTextChangedListener(watcher);
```
Comment (ajin12) [inline app/src/main/java/com/example/una/DonationActivity.java]: Nice refactoring

File: app/src/main/java/com/example/una/SurveyActivity.java [hunk 3]
```diff
@@ -113,6 +113,8 @@ public void onClick(View v) {
                     Map<String, Object> user = new HashMap<>();
                     user.put("first_name", firstName);
                     user.put("last_name", lastName);
+                    user.put("fav_charity_ein", R.string.red_cross_ein);
```
Comment (ajin12) [inline app/src/main/java/com/example/una/SurveyActivity.java]: Add TODO to replace this later and create new issue?

Comment (alexandrashaw10) [reply app/src/main/java/com/example/una/SurveyActivity.java]: Yeah. We can either do that, or pick one randomly based on their interests, or we can let them change it manually by setting a new charity as their favorite on that charity's detail page

**Comment by alexandrashaw10** [comment]
#57

File: app/build.gradle [hunk 4]
```diff
@@ -28,12 +28,13 @@ android {
...
 }
 
+
 dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
 
-    implementation 'com.google.firebase:firebase-database:18.0.0'
     implementation 'androidx.appcompat:appcompat:1.0.2'
     implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
+    implementation 'androidx.recyclerview:recyclerview:1.0.0'
```
Comment (chiped) [inline app/build.gradle]: thanks for grouping these together by functionality

File: app/build.gradle [hunk 5]
```diff
@@ -60,20 +61,25 @@ dependencies {
     // google login
     implementation 'com.google.firebase:firebase-auth:18.1.0'
     implementation 'com.google.android.gms:play-services-auth:17.0.0'
+    // classpath 'com.google.gms:google-services:4.2.0'
```
Comment (chiped) [inline app/build.gradle]: was this intended?

File: app/src/main/java/com/example/una/FirestoreClient.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+import java.text.DecimalFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+public class FirestoreClient {
```
Comment (chiped) [inline app/src/main/java/com/example/una/FirestoreClient.java]: 😍

File: app/src/main/java/com/example/una/FirestoreClient.java [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+    private final String TAG = "FirestoreClient";
+    private FirebaseFirestore db = FirebaseFirestore.getInstance();
+    private FirebaseUser user;
+    Context context;
+
+    public FirestoreClient(Context context) {
+        user = FirebaseAuth.getInstance().getCurrentUser();
```
Comment (chiped) [inline app/src/main/java/com/example/una/FirestoreClient.java]: could reuse `db` here

**Pull request merged**